types/stylelint/index.d.ts

```diff
@@ -243,6 +243,135 @@ declare namespace stylelint {
 		meta?: RuleMeta;
 	};
 
+	/** @internal */
+	type BuiltInRules = {
```

Comment:
question

I wonder if this can be refactored using `keyof` so that it doesn't have to be maintained because it would be generated automatically.

Reply:
I think maintaining the type is not so hard. If we had an inconsistency between the implementation and the type, the type checker (`tsc`) could detect it.